### ISSUE TITLE
Geoserver Errors serving WFS layers

### ISSUE THREAD

**Issue by jonescc** [aodn/content#233]
There are a number if errors being reported by geoserver-123 trying to server layers and data over the last week or so due to geoserver config not matching database relations:

Some of these issues look to have been addressed already, but reporting them here for completeness

```
ERROR [geoserver.ows] - org.geoserver.platform.ServiceException: The requested Style can not be used with this layer.  The style specifies an attribute of time_start and the layer is: imos:soop_asf_fmt_trajectory_map	at org.geoserver.wms.map.GetMapKvpRequestReader.checkStyle(GetMapKvpRequestReader.java:1215)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:437)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:84)	at org.geoserver.ow
ERROR [geoserver.ows] - org.geoserver.platform.ServiceException: The requested Style can not be used with this layer.  The style specifies an attribute of time_start and the layer is: imos:soop_co2_trajectory_map	at org.geoserver.wms.map.GetMapKvpRequestReader.checkStyle(GetMapKvpRequestReader.java:1215)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:437)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:84)	at org.geoserver.ows.Di
ERROR [geoserver.ows] - org.geoserver.platform.ServiceException: The requested Style can not be used with this layer.  The style specifies an attribute of time_start and the layer is: imos:soop_sst_dm_trajectory_map	at org.geoserver.wms.map.GetMapKvpRequestReader.checkStyle(GetMapKvpRequestReader.java:1215)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:437)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:84)	at org.geoserver.ows
ERROR [geoserver.ows] - org.geoserver.platform.ServiceException: The requested Style can not be used with this layer.  The style specifies an attribute of time_start and the layer is: imos:soop_sst_nrt_trajectory_map	at org.geoserver.wms.map.GetMapKvpRequestReader.checkStyle(GetMapKvpRequestReader.java:1215)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:437)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:84)	at org.geoserver.ow
ERROR [geoserver.ows] - org.geoserver.platform.ServiceException: The requested Style can not be used with this layer.  The style specifies an attribute of time_start and the layer is: imos:soop_tmv_trajectory_map	at org.geoserver.wms.map.GetMapKvpRequestReader.checkStyle(GetMapKvpRequestReader.java:1215)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:437)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:84)	at org.geoserver.ows.Di
ERROR [geoserver.ows] - org.geoserver.platform.ServiceException: The requested Style can not be used with this layer.  The style specifies an attribute of time_start and the layer is: imos:soop_trv_trajectory_map	at org.geoserver.wms.map.GetMapKvpRequestReader.checkStyle(GetMapKvpRequestReader.java:1215)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:437)	at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:84)	at org.geoserver.ows.Di
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: Newport for feature type aodn:coastal_watch_locations	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:162)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:173)	at org.geotools.filter.IsEqualsToIm
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:auv_trajectory_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.IsGreaterThanO
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:soop_asf_mt_trajectory_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.IsGrea
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:soop_co2_trajectory_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.IsGreater
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:soop_sst_dm_trajectory_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.IsGrea
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:soop_sst_nrt_trajectory_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.IsGre
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:srs_altimetry_timeseries_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.IsGr
ERROR [geoserver.ows] - org.geoserver.wfs.WFSException: Illegal property name: time_start for feature type imos:srs_oc_soop_rad_trajectory_map	at org.geoserver.wfs.GetFeature$2.visit(GetFeature.java:1184)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:159)	at org.geotools.filter.visitor.AbstractFilterVisitor.visit(AbstractFilterVisitor.java:208)	at org.geotools.filter.Is
ERROR [geotools.jdbc] - Failed to execute statement SELECT "channel_id","TIME","VALUES","VALUES_quality_control","url","platform_code","site_code","LATITUDE","LONGITUDE","NOMINAL_DEPTH","VARNAME","LONGNAME",encode(ST_AsEWKB("geom"),'base64') as "geom" FROM "anmn_nrs_dar_yon"."anmn_nrs_dar_yon_timeseries_data" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","abstract","url_marlin_metadata","cruise_id","data_type","url_nc","url_csv","url_plot","size","time_start","time_end","vessel_name","colour",encode(ST_AsEWKB("geom"),'base64') as "geom" FROM "srs_oc_bodbaw"."srs_oc_bodbaw_trajectory_profile_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","size","url","site_name","instrument","time_start","time_end","TEMP_b","PSAL_b","PRES_b","CNDC_b","UCUR_b","VCUR_b",encode(ST_AsBinary(CASE WHEN ST_HasArc("geom") THEN "geom" ELSE ST_Simplify(ST_Force_2D("geom"), 0.07031249919999709) END),'base64') as "geom" FROM "srs_altimetry"."srs_altimetry_timeseries_map" WHERE  "geom" && ST_GeomFromText('POLYGON ((146.38183586270628 -44.561523232, 146.38183586270628 -41.924804512, 149.0185
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","size","url","site_name","instrument","time_start","time_end","TEMP_b","PSAL_b","PRES_b","CNDC_b","UCUR_b","VCUR_b",encode(ST_AsBinary(CASE WHEN ST_HasArc("geom") THEN "geom" ELSE ST_Simplify(ST_Force_2D("geom"), 0.07031249919999709) END),'base64') as "geom" FROM "srs_altimetry"."srs_altimetry_timeseries_map" WHERE  "geom" && ST_GeomFromText('POLYGON ((155.96191392475248 -40.7822264, 155.96191392475248 -38.14550768, 158.5986326
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","size","url","site_name","instrument","time_start","time_end","TEMP_b","PSAL_b","PRES_b","CNDC_b","UCUR_b","VCUR_b",encode(ST_AsEWKB("geom"),'base64') as "geom" FROM "srs_altimetry"."srs_altimetry_timeseries_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","size","url","time_coverage_start","time_coverage_end","LATITUDE","LONGITUDE",encode(ST_AsEWKB("geom"),'base64') as "geom","instrument" FROM "srs_oc_ljco_wws"."srs_oc_ljco_wws_daily_all_products_fv02_timeseries_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","url","size","time_start","time_end","colour","CNDC_b","CPHL_b","PSAL_b","TEMP_b","TURB_b",encode(ST_AsEWKB("geom"),'base64') as "geom","platform_code","vessel_name" FROM "soop_tmv"."soop_tmv_trajectory_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","url","size","time_start","time_end","colour",encode(ST_AsEWKB("geom"),'base64') as "geom","platform_code","cruise_id","vessel_name" FROM "soop_asf_fmt"."soop_asf_fmt_trajectory_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","url","size","time_start","time_end","colour",encode(ST_AsEWKB("geom"),'base64') as "geom","platform_code","cruise_id","vessel_name" FROM "soop_asf_mt"."soop_asf_mt_trajectory_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","url","size","time_start","time_end","colour",encode(ST_AsEWKB("geom"),'base64') as "geom","platform_code","cruise_id","vessel_name" FROM "soop_co2"."soop_co2_trajectory_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "file_id","url","size","time_start","time_end","colour",encode(ST_AsEWKB("geom"),'base64') as "geom","platform_code","vessel_name" FROM "srs_oc_soop_rad"."srs_oc_soop_rad_trajectory_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "trajectory_id","url","platform_code","vessel_name","voyage_number","time_start","time_end",encode(ST_AsEWKB("geom"),'base64') as "geom","colour","AIRT_b","ATMP_b","CNDC_b","DEWT_b","GPS_HEIGHT_b","PL_CRS_b","PL_SPD_b","PL_WDIR_b","PL_WSPD_b","PSAL_b","RELH_b","TEMP_b","WDIR_b","WETT_b","WSPD_b","size" FROM "soop_sst"."soop_sst_nrt_trajectory_map" LIMIT 1
ERROR [geotools.jdbc] - Failed to execute statement SELECT "trajectory_id","url","platform_code","vessel_name","voyage_number","time_start","time_end",encode(ST_AsEWKB("geom"),'base64') as "geom","colour","AIRT_b","ATMP_b","GPS_HEIGHT_b","PL_CRS_b","PL_SPD_b","PL_WDIR_b","PL_WSPD_b","RAD_PAR_b","RELH_b","TEMP_b","WDIR_b","WSPD_b","size" FROM "soop_sst"."soop_sst_dm_trajectory_map" LIMIT 1
ERROR [geotools.rendering] - Property 'TIME' could not be found in soop_sst_nrt_trajectory_mapjava.lang.IllegalArgumentException: Property 'TIME' could not be found in soop_sst_nrt_trajectory_map	at org.geotools.filter.visitor.PostPreProcessFilterSplittingVisitor.visit(PostPreProcessFilterSplittingVisitor.java:787)	at org.geotools.filter.AttributeExpressionImpl.accept(AttributeExpressionImpl.java:340)	at org.geotools.filter.visitor.PostPreProcessFilterSplittingVisitor.visitBinaryComparisonOperat
```

**Comment by julian1** [comment]
According to @jonescc the issue is a change to the TIME variable name in the database schema being out of sync with the geoserver layer definition of that variable.

The database schema (defined in liquibase in the harvester), the geoserver layer and the geonetwork record all need to be versioned, tested, and promoted to production in parallel.

A good way to achieve that, would be to combine the geoserver, harvest schema, md record content repositories into a single repo. 

The advantage is that commit history can capture and version the content state between each app - and not just for each app. 

That would allow configuration consistency to be guaranteed - at the time of testing, when content is promoted to production, and when rolling back to some prior content state in order to reproduce and fix a bug.

**Comment by jonescc** [comment]
Still getting errors for imos:soop_asf_fmt_trajectory_map in yesterday and today's log

**Comment by jonescc** [comment]
Still getting errors for anmn_nrs_dar_yon_timeseries_data as well

**Comment by lbesnard** [comment]
> A good way to achieve that, would be to combine the geoserver, harvest schema, md record content repositories into a single repo.

ahhh, I'm hearing you !

**Comment by lbesnard** [comment]
should be fix by a restart of geoserver

**Comment by jonescc** [comment]
soop_asf_fmt_trajectory_map looks OK now.  GetCapabilities still failing rendering metadata for  anmn_nrs_dar_yon_timeseries_data but this is the subject of a separate issue https://github.com/aodn/content/issues/235
